wordpress-mobile/WordPress-Android
# Update SiteCreationService from latest AutoForeground

Bring over to SiteCreationService the latest AutoForeground updates from `develop`. While at it, unify and remove code duplication by moving common functionality inside the `AutoForeground` class.

To test #1:
* Make sure that the testplan in https://github.com/wordpress-mobile/WordPress-Android/pull/6878#totest still works OK

To test #2:
* Make sure that the testplan in https://github.com/wordpress-mobile/WordPress-Android/pull/6825#totest also works OK
